lib/dropwizard/auth/src/main/java/com/wikia/dropwizard/auth/APIGatewayAuthFactory.java

```
@@ -57,33 +79,27 @@ public void setRequest(HttpServletRequest request) {
   @SuppressWarnings("unchecked")
   public T provide() {
     if (request != null) {
-      final APIGatewayCredentials apiGatewayCredentials = APIGatewayCredentials.builder()
-          .userId(NumberUtils.toInt(request.getHeader(GATEWAY_USER_ID_HEADER), 0))
-          .build();
+      String header = request.getHeader(GATEWAY_USER_ID_HEADER);
 
       try {
-        final Optional<T> result = authenticator().authenticate(apiGatewayCredentials);
-        return result.or((T)ANONYMOUS_USER);
+        if (header != null) {
+          final APIGatewayCredentials apiGatewayCredentials = APIGatewayCredentials.builder()
+              .userId(NumberUtils.toInt(header))
+              .build();
+
+          final Optional<T> result = authenticator().authenticate(apiGatewayCredentials);
+          if (result.isPresent()) {
+            return result.get();
+          }
+        }
       } catch (AuthenticationException e) {
         LOGGER.warn("Error authenticating credentials", e);
+        throw new InternalServerErrorException();
```

It's mostly for hide all possible exceptions information (stacktrace etc.) from exception that can be showed to user.